Scripts/generate-test-config.sh

```diff
@@ -1,14 +1,269 @@
+  esac
+done
+
+
+##################################################
+# VALIDATIONS AND ASSIGNMENTS
+
+cmd_quiet_flag="--quiet"
+if [[ $LOG_LEVEL -gt 1 ]] ; then
+  cmd_quiet_flag=""
+fi
+
+if [[ -z $platform ]] ; then
+  die "'platform' not specified"
+fi
+
+if [[ -z $region ]] ; then
+  die "'region' not specified"
+fi
+
+python_cmd=$(which python3)
+if [[ -z $python_cmd ]] ; then
+  die "Cannot find 'python3' in PATH"
+fi
+
+if [[ $assume_role ]] ; then
+  aws_cli_cmd=$(which aws)
+  if [[ -z $aws_cli_cmd ]] ; then
+    die "Cannot find 'aws' in PATH"
+  fi
+
+  jq_cmd=$(which jq)
+  if [[ -z $jq_cmd ]] ; then
+    die "Cannot find 'jq' in PATH"
+  fi
+fi
+
+
+##################################################
+# MAIN
+
+# Exit if any command fails. Don't set this flag until after argument and
+# environment validation, or commands like `which foo` that test for
+# dependencies will cause an immediate failure with no error message
+set -e
+
+trap exit SIGINT
+
+log_info "Generating configuration file for ${platform}"
+
+starting_dir=$PWD
+
+# Set up output paths
+dest_dir=$HOME/.aws-amplify/aws-sdk-${platform}
+dest_file=${dest_dir}/testconfiguration.json
+mkdir -p $dest_dir
+
+# Set up support repo
+support_repo_name=amplify-ci-support
+support_repo_branch=palpatim/build_testconfig
+support_repo_url=https://github.com/aws-amplify/${support_repo_name}.git
+
+cd $dest_dir
+
+# Install CI support repo
+
+if [[ -d ${dest_dir}/${support_repo_name} ]] ; then
+  log_debug "Support repo exists at '${dest_dir}/${support_repo_name}. Fetching latest version."
+  cd $support_repo_name
+  git fetch origin $cmd_quiet_flag
+  git checkout -B $support_repo_branch $cmd_quiet_flag
+else
+  log_debug "Cloning support repo into '${dest_dir}/${support_repo_name}"
+  git clone $support_repo_url --branch $support_repo_branch $support_repo_name $cmd_quiet_flag
+  cd $support_repo_name
 fi
 
-TEMP_FILE=$(mktemp aws-sdk-ios.testconfiguration.json)
-echo '{"credentials":{"accessKey":"EMPTY","secretKey":"EMPTY","sessionToken":"EMPTY"},"packages":{"common":{"region":"us-west-2"}}}' > $TEMP_FILE
-cp $TEMP_FILE $DESTINATION
-rm $TEMP_FILE
+# Install dependencies
+cd ./src/integ_test_resources/common/scripts
+log_debug "Setting up and activating python virtual environment"
+$python_cmd -m venv .env
+source ./.env/bin/activate $cmd_quiet_flag
+
+log_debug "Installing python requirements"
+pip3 install --upgrade pip $cmd_quiet_flag
+pip3 install -r requirements.txt $cmd_quiet_flag
+
+# Assume test execution role and export those values so the builder script can pick them up from the environment
+if [[ $assume_role ]] ; then
+  log_debug "Assuming test execution IAM role"
+  circleci_execution_role_arn=$(aws ssm get-parameter --name '/mobile-sdk/ios/common/circleci_execution_role' | jq -r .Parameter.Value)
+  assume_role_creds=$(aws sts assume-role --duration-seconds 3600 --role-arn "${circleci_execution_role_arn}" --role-session-name "IntegTest-$(date +%Y%m%d%H%M%S)")
```

Comment:
Does this particular `date` command work accross `Darwin` and `Linux` `uname`s?

Reply:
In my experience, yes; the `+` with normal `strftime` symbols seems to be pretty universal. I've had mixed results with extensions like "natural language" date math ("today", "7 days ago"), which are supported on some variants and not others, but I've never had problems with simple timestamp formats of the current date.

Reply:
Seems to, FWIW:

*macOS*
```console
$ date +%Y%m%d%H%M%S   
20200506103905
```

*Linux*
```console
$ date +%Y%m%d%H%M%S
20200506143909
```